### PR TITLE
Pbs batch fixes

### PR DESCRIPTION
**Pull request by jedwards4b**
Fixes issues with spaces in the qsub command line on pbs systems.
Fixes some issues with user_mods support.

Test suite: 
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes: 

User interface changes?: 

Code review:

### CODE REVIEW

File: cime_config/cesm/machines/config_machines.xml [hunk 1]
```diff
@@ -868,7 +868,10 @@
     <DESC>"NCAR CGD Linux Cluster 48 pes/node, batch system is PBS"</DESC>
     <NODENAME_REGEX>hobart</NODENAME_REGEX>
     <COMPILERS>intel,pgi,nag,gnu</COMPILERS>
-    <MPILIBS>mvapich2,openmpi,mpi-serial</MPILIBS>
```
Comment (jedwards4b) [inline cime_config/cesm/machines/config_machines.xml]: mpi-serial is always valid and not needed,  Added support for different compilers to support a different set of mpilibs

File: driver_cpl/cime_config/config_component.xml [hunk 2]
```diff
@@ -643,7 +643,7 @@
     <file>env_build.xml</file>
     <desc>Operating system - DO NOT EDIT UNLESS for userdefined machine - ignored once Macros has been created.</desc>
   </entry>
-
+<!--
```
Comment (jedwards4b) [inline driver_cpl/cime_config/config_component.xml]: I don't think that we need COMPILERS and MPILIBS in the case.  I've commented them out, but can remove them if you prefer.

File: utils/python/CIME/XML/grids.py [hunk 3]
```diff
@@ -27,6 +27,19 @@ def get_grid_info(self, name, compset):
         """
         nodes = self.get_nodes("grid")
         gridinfo = {}
+        atmnlev = None
```
Comment (jedwards4b) [inline utils/python/CIME/XML/grids.py]: Grid specification can optionally include vertical grid info for clm and cam.  Add support for that feature here.

File: utils/python/CIME/XML/machines.py [hunk 4]
```diff
@@ -133,7 +133,7 @@ def set_machine(self, machine):
 
         return machine
 
-    def get_value(self, name, attributes={}, resolved=True, subgroup=None):
+    def get_value(self, name, attributes=None, resolved=True, subgroup=None):
```
Comment (jedwards4b) [inline utils/python/CIME/XML/machines.py]: null setting for attributes was inconsistent.

File: utils/python/CIME/batch_utils.py [hunk 5]
```diff
@@ -120,9 +120,10 @@ def get_submit_args(self, job):
             if name is None:
                 submitargs+=" %s"%flag
             else:
-                val = self.case.get_value(name,subgroup=job)
+                val = self.case.get_value(name,subgroup=job).strip()
                 if val is not None and len(val) > 0 and val != "None":
-                    if flag.rfind("[:=]", len(flag)-1, len(flag)) >= 0:
```
Comment (jedwards4b) [inline utils/python/CIME/batch_utils.py]: bug i introduced.

File: utils/python/CIME/SystemTests/pem.py [hunk 6]
```diff
@@ -31,7 +31,7 @@ def build(self, sharedlib_only=False, model_only=False):
             logging.warn("Starting bld %s"%bld)
             self._case.flush()
             case_setup(self._caseroot,reset=True)
-            run_cmd('case.build --clean')
+            run_cmd('./case.build --clean')
```
Comment (wilke) [inline utils/python/CIME/SystemTests/pem.py]: Shouldn't you add  from_dir=self._caseroot to  be consistent with run_cmd calls in pea.py , ncr.py , ....

File: utils/python/CIME/SystemTests/pet.py [hunk 7]
```diff
@@ -25,7 +25,7 @@ def build(self, sharedlib_only=False, model_only=False):
 
         self._case.flush()
         case_setup(self._caseroot, reset=True)
-        run_cmd('case.build --clean')
+        run_cmd('./case.build --clean')
```
Comment (wilke) [inline utils/python/CIME/SystemTests/pet.py]: Add  from_dir=self._caseroot ?

File: cime_config/cesm/machines/config_machines.xml (hunk 1 above)
Comment (jgfouca) [inline cime_config/cesm/machines/config_machines.xml]: Is this change backwards compatible? Meaning we don't have to change mpilibs for all machines?

File: driver_cpl/cime_config/config_component.xml [hunk 8]
```diff
@@ -644,16 +644,6 @@
     <desc>Operating system - DO NOT EDIT UNLESS for userdefined machine - ignored once Macros has been created.</desc>
   </entry>
 
-  <entry id="COMPILERS">
```
Comment (jgfouca) [inline driver_cpl/cime_config/config_component.xml]: Why were these removed?

File: cime_config/cesm/machines/config_machines.xml (hunk 1 above)
Comment (jedwards4b) [inline cime_config/cesm/machines/config_machines.xml]: Yes

File: driver_cpl/cime_config/config_component.xml (hunk 8 above)
Comment (jedwards4b) [inline driver_cpl/cime_config/config_component.xml]: My change to allow multiple MPILIBS on a machine was causing an issue in parsing the machines file to write the env_build file - these variables are not used in a case so the best solution was not to write them there.

**Comment by jgfouca** [comment]
Ready to merge pending @wilke 's requested fixes.

**Comment by jedwards4b** [comment]
I've addressed @wilke's concern by replacing the run_cmd calls with a python subroutine call.
